src/integrationTest/java/com/mongodb/hibernate/service/MongoIdFieldNameIntegrationTests.java

```diff
@@ -0,0 +1,142 @@
+@DomainModel(
+        annotatedClasses = {
+            MongoIdFieldNameIntegrationTests.EntityWithoutIdColumnAnnotation.class,
+            MongoIdFieldNameIntegrationTests.EntityWithIdColumnAnnotationWithoutNameProperty.class,
+            MongoIdFieldNameIntegrationTests.EntityWithIdColumnAnnotationWithNamePropertyIdentical.class
+        })
+class MongoIdFieldNameIntegrationTests implements SessionFactoryScopeAware {
+
+    @AutoClose
+    private MongoClient mongoClient;
+
+    private MongoCollection<BsonDocument> collection;
+
+    private SessionFactoryScope sessionFactoryScope;
+
+    @Override
+    public void injectSessionFactoryScope(SessionFactoryScope sessionFactoryScope) {
+        this.sessionFactoryScope = sessionFactoryScope;
+    }
+
+    @BeforeAll
+    void beforeAll() {
+        var config = new MongoConfigurationBuilder(
+                        sessionFactoryScope.getSessionFactory().getProperties())
+                .build();
+        mongoClient = MongoClients.create(config.mongoClientSettings());
+        collection = mongoClient.getDatabase(config.databaseName()).getCollection("movies", BsonDocument.class);
+    }
+
+    @BeforeEach
+    void beforeEach() {
+        collection.drop();
+    }
+
+    @Test
+    void testEntityWithoutIdColumnAnnotation(SessionFactoryScope scope) {
+        scope.inTransaction(session -> {
+            var movie = new EntityWithoutIdColumnAnnotation();
+            movie.id = 1;
+            session.persist(movie);
+        });
+    }
+
+    @Test
+    void testEntityWithIdColumnAnnotationWithoutNameProperty(SessionFactoryScope scope) {
+        scope.inTransaction(session -> {
+            var movie = new EntityWithIdColumnAnnotationWithoutNameProperty();
+            movie.id = 1;
+            session.persist(movie);
+        });
+    }
+
+    @Test
+    void testEntityWithIdColumnAnnotationWithNamePropertyIdentical(SessionFactoryScope scope) {
+        scope.inTransaction(session -> {
+            var movie = new EntityWithIdColumnAnnotationWithNamePropertyIdentical();
+            movie.id = 1;
+            session.persist(movie);
+        });
+    }
+
+    // TO-DO-HIBERNATE-56 https://jira.mongodb.org/browse/HIBERNATE-56
+    // add testing case for the case that id column is explict specified and different from '_id'
+
+    @Entity(name = "Entity1")
+    @Table(name = "movies")
+    static class EntityWithoutIdColumnAnnotation {
+        @Id
+        int id;
+    }
+
+    @Entity(name = "Entity2")
+    @Table(name = "movies")
+    static class EntityWithIdColumnAnnotationWithoutNameProperty {
+        @Id
+        @Column
+        int id;
+    }
+
+    @Entity(name = "Entity4")
+    @Table(name = "movies")
+    static class EntityWithIdColumnAnnotationWithNamePropertyIdentical {
+        @Id
+        @Column(name = "_id")
+        int id;
+    }
+
+    private void onMongoCollection(String collectionName, Consumer<MongoCollection<BsonDocument>> collectionConsumer) {
```

Comment:
Not sure what you're intention is here, as this method is never called.

Reply:
yeah, it is leftover from my previous code result. Deleted.